mdn/kuma
# better test coverage on stripe utils

Fixes #6916

How to test it:
```
pytest --cov=kuma --no-cov-on-fail kuma/api/v1/tests/test_views.py -x --ff && coverage html
```
then `open htmlcov/kuma_users_stripe_utils_py.html` and see that it's mostly covered. 

This turned out to be a lot messier than I had hoped. And we are potentially taking on some duplication of tests. 

The point is, the v1 JSON API is the only "API" that matters. What these tests are trying to simulate is a human QA person trying to do things like listing subscriptions, canceling subscriptions, and creating subscriptions all using `curl`. That QA person wouldn't care how it's done as long as it doesn't actually use Stripe over the network. 

The mocking got a little hairy and it's as fragile as all mocking always is. But it walks important business logic code we've written. 

Now, I fear we still have a bunch of duplicate tests somewhere in `kuma.payments` and `kuma.users` but those will soon *not* have a public API. Once we get rid of subscription stuff on the Edit Profile page, we can tackle more refactoring. That day (when we get rid of subscriptions on the Edit Profile, and the Subscription management page has been rewritten in React) then we can revisit some tests in `users/tests/*.py` and `payments/tests/*.py`.
